creates script to send API calls hugging face models

aaroosh-07 opened the pull request:
this PR creates a python script to communicate with hugging face models via post request to perform question answering task on a given context.

aaroosh-07 committed:
creates script to hugging face models

prattyushmangal commented:
You need to add a readme first of all describing what this repo is and how you can run this script. 

Also is this marking the decision to go with HF instead of OpenAI? This pr or the ticket should address the conversations to resolve in the issue with a ADR (Architecture Decision Record markdown) too.

prattyushmangal commented:
You also need to provide a venv and packages guide via a requirements.txt to enable people to recreate your environment and run your script.

aaroosh-07 committed:
updates readme and adds requirements.txt

aaroosh-07 commented:
I have added requirements.txt file to show all the modules needed to run the script and also updated readme file to include a small description of the project and steps to install all dependencies and run the script

aaroosh-07 commented:
Yes we are going ahead with huggingFace models as OpenAI provide with limited free credits and after that we will need to purchase the credits. In developement phase we can use hugging face models to build the application and then add support for OpenAI separately

prattyushmangal commented on README.md:
````diff
@@ -1 +1,47 @@
+python -m venv virtualenv
+```
+
+2. Install all the required modules.
````
You need to activate the virtual environment before you can install the requirements. If you install the requirements before you start you env then the installations will take place in your global python installation which is definitely not what you want to do.

aaroosh-07 replied:
I will add a step to activate virtual environment before installing requirements.

prattyushmangal reviewed (changes requested):
The main script is fine, but we should also add a script which uses the HF SDK instead of directly using the REST API from HF. Using the SDK will give us access to greater functionality and exception handling.

prattyushmangal commented:
Also lets put these experimental scripts in an `experiments` directory or something.

aaroosh-07 commented:
> The main script is fine, but we should also add a script which uses the HF SDK instead of directly using the REST API from HF. Using the SDK will give us access to greater functionality and exception handling.

There is only transformers SDK provided by HF, which allows to download models and run them locally, but no SDK is present to use models deployed on serverless infrastructure by HF and we can only prompt them by sending REST API calls.

aaroosh-07 committed:
resolves review comments

prattyushmangal commented on README.md:
```diff
@@ -1 +1,57 @@
+
+### Running the script
+
+1. Activate the virtual environment
```
Why is this here if this is already in the Setting up the environment section?

prattyushmangal reviewed (approved):
Little things - pay closer attention to detail, but tackle this in the next pull request.
